test/jdk/java/net/DatagramSocket/DatagramSocketMulticasting.java

```diff
@@ -298,23 +307,29 @@ static void testSendReceive(DatagramSocket s, InetAddress group) throws IOExcept
         System.out.println("testSendReceive");
 
         // outgoing multicast interface needs to be set
-        assertTrue(s.getOption(IP_MULTICAST_IF) != null);
+        assertNotNull(s.getOption(IP_MULTICAST_IF));
 
         SocketAddress target = new InetSocketAddress(group, s.getLocalPort());
-        byte[] message = "hello".getBytes("UTF-8");
+        byte[] message = "testSendReceive".getBytes("UTF-8");
 
         // send message to multicast group
         DatagramPacket p = new DatagramPacket(message, message.length);
         p.setSocketAddress(target);
         s.send(p);
 
-        // receive message
+        // receive message with retry in case of stray messages
         s.setSoTimeout(0);
-        p = new DatagramPacket(new byte[1024], 100);
-        s.receive(p);
+        for (int i = 1; i <= MAX_TRIES; i++) {
+            p = new DatagramPacket(new byte[1024], 100);
+            s.receive(p);
+            if(s.getLocalPort()==p.getPort()){
+                assertEquals(p.getLength(), message.length,
+                        String.format("expected message %s, instead received %s%n", message, p));
+                break;
+            }
 
-        assertTrue(p.getLength() == message.length);
-        assertTrue(p.getPort() == s.getLocalPort());
+            assertNotEquals(i, MAX_TRIES, "testSendReceive: too many retries");
```

Comment:
It would be good to log the content of the received datagram (address of the sender, + content as text) for debugging purpose. possibly the logging could be done whether the datagram matches or not. and maybe the interation number (i) could be included in the log message.